### PR TITLE
feat: deferred voting

### PR DESCRIPTION
**Pull request by pauldelucia**
Allow users to schedule masternode votes to be executed later. The app must be running in order for them to be executed automatically.

After clicking on a vote choice on the DPNS Active Contests screen and selecting a node (or all), there is an option to vote now or later. If later is selected, they are given a form to select a time.

There is also a new DPNS subscreen for Scheduled Votes where you can view all scheduled votes and have the option to remove them or cast an individual one immediately instead of at the scheduled time (may be useful if the app is closed when it was supposed to be executed).

## Release Notes

- **New Features**
  - Introduced functionality for managing scheduled votes, including a new screen for scheduling votes.
  - Added a new subscreen option for scheduled votes in the user interface.
  - Enhanced the `DPNSContestedNamesScreen` to display and manage scheduled votes.
  
- **Bug Fixes**
  - Improved error handling and messaging related to voting processes.

- **Documentation**
  - Updated various methods and enums to reflect new functionalities related to scheduled votes.

- **Chores**
  - Updated the database initialization process to support a new version for scheduled votes.

These enhancements significantly improve the user experience with voting tasks and scheduled votes management.

### CODE REVIEW

File: src/database/scheduled_votes.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            };
+            let vote_choice = match vote_choice_string.as_str() {
+                "Abstain" => ResourceVoteChoice::Abstain,
+                "Lock" => ResourceVoteChoice::Lock,
+                other => {
+                    if let Some(inner) = other.strip_prefix("TowardsIdentity(") {
+                        if let Some(inner) = inner.strip_suffix(')') {
+                            let towards_id = inner.to_string();
+                            ResourceVoteChoice::TowardsIdentity(
+                                Identifier::from_string(&towards_id, Encoding::Base58)
```
Comment (pauldelucia) [inline src/database/scheduled_votes.rs:103]: Check: is it always going to be Base58?

Comment (pauldelucia) [reply src/database/scheduled_votes.rs:103]: Also: is this the best way to store vote choices?

File: src/ui/dpns_contested_names_screen.rs [hunk 2]
```diff
@@ -61,11 +66,13 @@ pub struct DPNSContestedNamesScreen {
     user_identities: Vec<QualifiedIdentity>,
     contested_names: Arc<Mutex<Vec<ContestedName>>>,
     local_dpns_names: Arc<Mutex<Vec<(Identifier, DPNSNameInfo)>>>,
+    scheduled_votes: Arc<Mutex<Vec<ScheduledDPNSVote>>>,
```
Comment (pauldelucia) [inline src/ui/dpns_contested_names_screen.rs:69]: I'm curious if these all need to be mutexes

File: src/ui/dpns_contested_names_screen.rs [hunk 3]
```diff
@@ -232,28 +247,44 @@ impl DPNSContestedNamesScreen {
...
+                            }
+                            _ => {
+                                // To Do: Some kind of refresh for scheduled votes maybe
+                                app_action |= AppAction::None;
+                            }
                         }
                     }
                 }
+            } else {
+                ui.label("Go to the Active Contests subscreen to schedule votes.");
```
Comment (pauldelucia) [inline src/ui/dpns_contested_names_screen.rs:287]: Can this be cleaned up?

File: src/ui/dpns_contested_names_screen.rs [hunk 4]
```diff
@@ -688,6 +719,176 @@ impl DPNSContestedNamesScreen {
         });
     }
 
+    fn render_table_scheduled_votes(&mut self, ui: &mut Ui) -> AppAction {
```
Comment (pauldelucia) [inline src/ui/dpns_contested_names_screen.rs:722]: Fix sorting

**Comment by pauldelucia** [comment]
@coderabbitai review

**Review by ogabrielides — APPROVED** [review]
LGTM

**Pull request merged**